### PR TITLE
disallow people from using bad names

### PR DESCRIPTION
**Pull request by olliy1or**

### CODE REVIEW

**Comment by LoneWolfHT** [comment]
Looks like you're using spaces instead of tabs for indentation?

**Comment by olliy1or** [comment]
> Looks like you're using spaces instead of tabs for indentation?

i just hit enter at the end of other line :thinking:

File: disallowed_names/init.lua [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        if string.find(name:lower(), dn:lower()) then
+            return "You cannot use that name on this server. Please try a different name."
+        end
+    end
+end)
+
+-- Command to add a name to disallowed names
+minetest.register_chatcommand("bdname_add", {
+    params = "<name>",
+    privs = { ban = true },
```
Comment (Panquesito7) [inline disallowed_names/init.lua]: Should we allow guardians too (kick)?

File: disallowed_names/init.lua [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+			storage:set_string("disallowed_names", minetest.serialize(disallowed_names))
+
+			return true, "Added " .. param .. " to the list of banned names"
+		else
+			return false, "You need to provide a name to add"
+		end
+    end
+})
+
+--removes a name from disallowed names
```
Comment (Panquesito7) [inline disallowed_names/init.lua]:
```suggestion
-- removes a name from disallowed names
```

File: disallowed_names/init.lua [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        if param ~= "" then
+            for k in pairs(disallowed_names) do
+                if param == disallowed_names[k] then
+                    table.remove(disallowed_names, k)
+                end
+            end
+            storage:set_string("disallowed_names", minetest.serialize(disallowed_names))
+			return true, "Removed" .. param .. "from the list of banned names"
+        else
+			return false, "You need to provide a name to remove"
```
Comment (Panquesito7) [inline disallowed_names/init.lua]:
```suggestion
			return false, "You need to provide a name to remove."
```

File: disallowed_names/init.lua [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    privs = { ban = true },
+    func = function(name, param)
+        if param ~= "" then
+            for k in pairs(disallowed_names) do
+                if param == disallowed_names[k] then
+                    table.remove(disallowed_names, k)
+                end
+            end
+            storage:set_string("disallowed_names", minetest.serialize(disallowed_names))
+			return true, "Removed" .. param .. "from the list of banned names"
```
Comment (Panquesito7) [inline disallowed_names/init.lua]:
```suggestion
			return true, "Removed " .. param .. " from the list of banned names."
```

File: disallowed_names/init.lua [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+			return true, "Added " .. param .. " to the list of banned names"
+		else
+			return false, "You need to provide a name to add"
+		end
+    end
+})
+
+--removes a name from disallowed names
+minetest.register_chatcommand("bdname_remove",{
+    description = "removes a name from disallowed names",
```
Comment (Panquesito7) [inline disallowed_names/init.lua]:
```suggestion
    description = "Removes a name from disallowed names",
```

File: disallowed_names/init.lua [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    privs = { ban = true },
+    description = "Adds a name to the disallowed names list.",
+    func = function(name,param)
+		if param ~= "" then
+			table.insert(disallowed_names, tostring(param))
+			storage:set_string("disallowed_names", minetest.serialize(disallowed_names))
+
+			return true, "Added " .. param .. " to the list of banned names"
+		else
+			return false, "You need to provide a name to add"
```
Comment (Panquesito7) [inline disallowed_names/init.lua]:
```suggestion
			return false, "You need to provide a name to add."
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Maybe provide a `README.md` explaining what the mod does, how does it work, etc..

**Comment by Thomas--S** [comment]
To be honest, I don't like this implementation because of two reasons:

1. I don't think that mentioning all default bad words in the source code is a good idea. I know that this might be a valid approach from a technical point of view, but I think that source code should not contain language that we forbid on the server.
2. As far as I can see from a quick look, this implementation will lead to many false-positives. For example, the German word "Fuchs" (meaning "fox") would be forbidden with your current blocklist. Similiar things will probably happen with other words, too.

**Comment by LoneWolfHT** [comment]
I guess this can only cover obviously bad names then

File: disallowed_names/init.lua (hunk 1 above)
Comment (olliy1or) [reply disallowed_names/init.lua]: change of plans (code)

**Comment by olliy1or** [comment]
I've updated it now it uses the words from the chat filter mod the server use, but i'm thinking does this really need a new mod? lol only 5lines

File: disallowed_names/depends.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+filter
```
Comment (Panquesito7) [inline disallowed_names/depends.txt:1]: Please, don't use `depends.txt`, it is deprecated, use `mod.conf` instead.

Comment (LoneWolfHT) [reply disallowed_names/depends.txt:1]: I added a mod.conf after merging but yes, depends.txt should be avoided unless you want 0.4.x compat

Comment (Panquesito7) [reply disallowed_names/depends.txt:1]: CTF doesn't support 0.4-series anymore anyways. 🙂

**Pull request merged**